tests/oclc_/test_identifier_lookup_coverage_provider.py

```diff
@@ -0,0 +1,247 @@
+        self.call_count = 0
+        self.called_with = []
+
+    def parse(self, db, tree, identifiers):
+        self.call_count += 1
+        self.called_with.append(identifiers)
+
+class MockParserSingle(MockParser):
+    def initial_look_up(self, db, tree):
+        return 2, []
+
+class MockParserMulti(MockParser):
+    def initial_look_up(self, db, tree):
+        results = []
+        owi_numbers = ["48446512", "48525129"]
+        for number in owi_numbers:
+            data = IdentifierData(Identifier.OCLC_WORK, number)
+            results.append(data)
+        return 4, results
+
+class MockProvider(IdentifierLookupCoverageProvider):
+
+    def _single(self, db, tree, identifier):
+        self.called_with = dict(tree=tree, identifier=identifier)
+
+    def _multiple(self, db, owi_data, identifier):
+        self.called_with = dict(owi_data=owi_data, identifier=identifier)
+
+class MockProviderSingle(MockProvider):
+    def _get_tree(self, isbn):
+        xml = sample_data("single_work_with_isbn.xml", "oclc_classify")
+        return etree.fromstring(xml, parser=etree.XMLParser(recover=True))
+
+class MockProviderMulti(MockProvider):
+    def _get_tree(self, isbn):
+        xml = sample_data("multi_work_with_owis.xml", "oclc_classify")
+        return etree.fromstring(xml, parser=etree.XMLParser(recover=True))
+
+class TestIdentifierLookupCoverageProvider(DatabaseTest):
+
+    SINGLE_ISBN = "9781620281932"
+    MULTI_ISBN = "0345391837"
+
+    def _tree(self, type):
+        if type == "single":
+            return MockProviderSingle(self._default_collection)._get_tree(self.SINGLE_ISBN)
+        else:
+            return MockProviderMulti(self._default_collection)._get_tree(self.MULTI_ISBN)
+
+    def _id(self, type):
+        if type == "single":
+            return self._identifier(Identifier.ISBN, self.SINGLE_ISBN)
+        else:
+            return self._identifier(Identifier.ISBN, self.MULTI_ISBN)
+
+    def test_process_item_single(self):
+        # Testing that, when process_item finds out that a document's status code is 2,
+        # it calls _single, passes in the correct tree and identifier as arguments,
+        # and returns the original ISBN.  Uses mocked versions of _get_tree,
+        # initial_look_up, and _single.
+
+        provider = MockProviderSingle(self._default_collection)
+        provider.parser = MockParserSingle()
+        id = self._id("single")
+
+        result = provider.process_item(id)
+        eq_(etree.tostring(provider.called_with["tree"]), etree.tostring(provider._get_tree(self.SINGLE_ISBN)))
+        eq_(provider.called_with["identifier"], id)
+        eq_(result, id)
+
+    def test_process_item_multi(self):
+        # Testing that, when process_item finds out that a document's status code is 4,
+        # it calls _multiple, passes in the correct OWIs, and
+        # returns the original ISBN.  Uses mocked versions of _get_tree, initial_look_up,
+        # and _multiple.
```

Comment:
You also need to test that if `_multiple` returns a list of things, `self._apply()` is called on each one with the origina identifier.

Reply:
This doesn't need to go into `test_process_item_single` or `test_process_item_multi`; it can be a separate test to avoid clutter. We'll go over different strategies for doing this later, but I think the best way is to create a method like `MockProvider.queue` so that we can queue up the result we want, and change `_single` and `_multiple` to return the next thing off the queue.